File: src/main/java/ru/serdtsev/homemoney/dto/HmResponse.java
package ru.serdtsev.homemoney.dto;

public class HmResponse implements java.io.Serializable {
  public String status;
  public Object data;

  @SuppressWarnings("unused")
  public HmResponse() {
  }

  private HmResponse(String status, Object data) {
    this.status = status;
    this.data = data;
  }

  public static HmResponse getOk() {
    return getOk(null);
  }

  public static HmResponse getOk(Object data) {
    return new HmResponse("OK", data);
  }

  public static HmResponse getFail(String status) {
    return getFail(status, null);
  }

  public static HmResponse getFail(String status, Object data) {
    return new HmResponse(status, data);
  }

}

File: src/main/java/ru/serdtsev/homemoney/dto/HmResponse.kt
package ru.serdtsev.homemoney.dto

class HmResponse(val status: String, val data: Any?) {
  companion object {
    fun getOk(data: Any? = null): HmResponse {
      if (data is PagedList<*>) {
        System.out.println("getOk for PagedList")
      }
      return HmResponse("OK", data)
    }

    @JvmOverloads
    fun getFail(status: String, data: Any? = null) = HmResponse(status, data)
  }
}
